src/main/java/org/cyclonedx/util/deserializer/ComponentWrapperDeserializer.java

```diff
@@ -72,10 +77,25 @@ public ComponentWrapper deserialize(
         return null;
     }
 
-    Component[] components = parser.readValueAs(Component[].class);
-
-    wrapper.setComponents(Arrays.asList(components));
-
+    List<Component> components = Collections.emptyList();
```

Comment:
As far as I can determine, the issue arises due to `MetadataDeserializer` using the shortcut of reading the tree from the XML (this is only an XML issue) token stream, then using a unconfigured `ObjectMapper` to convert to the target types. The context is then lost (e.g. XML parser, XML deserialization context) within child deserializers and we enter this deserializer re-parsed then tree now as plain JSON tokens with a vanilla JSON parser.

This list of components (when nested in metadata/../ancestors etc) might be absent (no tokens), a single object, or an array of objects. Fix here is to explicitly handle these cases. 

Not 100% happy - still a hack on top of the existing hack to handle component object lists. 🤷